pkg/apis/awsproviderconfig/v1alpha1/awsclusterproviderconfig_types.go

```diff
@@ -0,0 +1,74 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// AWSClusterProviderConfigSpec defines the desired state of AWSClusterProviderConfig
+type AWSClusterProviderConfigSpec struct {
+	// The AWS Region the cluster lives in.
+	Region string `json:"region"`
+
+	// SSHKeyName is the name of the ssh key to attach to the bastion host.
+	SSHKeyName string `json:"sshKeyName,omitempty"`
+}
+
+// AWSClusterProviderConfigStatus defines the observed state of AWSClusterProviderConfig
+type AWSClusterProviderConfigStatus struct {
```

Comment:
I'm not sure we should overload the providerconfig to include a persisted status here...

Reply:
oh hmm because the Cluster would have a status already?

Reply:
We'd be conflating spec/status concepts and would be splitting our status between the actual Cluster/Machine Status and Spec.ProviderConfig